scene/3d/collision_object_3d.cpp

```diff
@@ -110,6 +111,41 @@ void CollisionObject3D::_update_pickable() {
 	}
 }
 
+void CollisionObject3D::_update_debug_shapes() {
+	for (Set<uint32_t>::Element *shapedata_idx = debug_shapes_to_update.front(); shapedata_idx; shapedata_idx = shapedata_idx->next()) {
+		if (shapes.has(shapedata_idx->get())) {
+			ShapeData &shapedata = shapes[shapedata_idx->get()];
+			for (int i = 0; i < shapedata.shapes.size(); i++) {
+				ShapeData::ShapeBase s = shapedata.shapes[i];
+				if (s.debug_shape) {
+					s.debug_shape->queue_delete();
+				}
+				if (s.shape.is_null() || shapedata.disabled) {
+					return;
+				}
+
+				Ref<Mesh> mesh = s.shape->get_debug_mesh();
+				MeshInstance3D *mi = memnew(MeshInstance3D);
+				mi->set_transform(shapedata.xform);
+				mi->set_mesh(mesh);
+				add_child(mi);
```

Comment:
Please add `mi->force_update_transform();` after `add_child(mi)` in order to make sure the transform is correctly propagated before rendering. Right now it's wrong for one frame because it's set during a deferred call and the transform from CollisionObject3D gets propagated to the child only on the next physics tick.